### ISSUE TITLE
User profile pages return a 5xx if the username starts with a 0

### ISSUE THREAD

**Issue by diox** [mozilla/addons-frontend#6007]
Copied over from https://github.com/mozilla/addons-server/issues/9137 which was filed by @bqbn :

### Describe the problem and steps to reproduce it:
Looking at our daily access logs, I'm surprised to see that 50% of the 5xx are caused by people access users' profile pages. For instance, on 8/10/2018 we had 173 5xx on one proxy instance, and 86 of them happened when people accessing `/<locale>/users/<user_id>` endpoint.

And particularly, [1] accounted for 40% of all 5xx, or 69 of them. The `0xy9en` user seems to have a few popular addons (for example [2] and [3]) and thus its user profile page attracted many hits, but unfortunately they all ended up with 5xx.

[1] https://addons.mozilla.org/en-US/firefox/user/0xy9en/
[2] https://addons.mozilla.org/en-US/firefox/addon/hotspot-shield-free-vpn-proxy-/
[3] https://addons.mozilla.org/en-US/firefox/addon/touch-vpn/?src=search

### What happened?

Not sure. A full list of user profile endpoints that returned 5xx is following:

```
/tr/firefox/user/000mayte000/
/fr/firefox/user/00Thathy00/
/en-GB/android/user/0SuperNova0/
/en-US/firefox/user/0owesleyo0/
/bg/firefox/user/0Adrian0Avalos0/
/tr/firefox/user/0_0_HellishHeaven_In_Flesh123456/
/ru/firefox/user/0veR/
/ru/firefox/user/000MIIX/
/ru/firefox/user/0x6b/
/nl/firefox/user/00adef/
/en-GB/firefox/user/0o0rsi/
/en-US/firefox/user/0leg-cool/
/en-US/firefox/user/0xy9en/
/fa/firefox/user/0xy9en/
/hsb/firefox/user/0xy9en/
/id/firefox/user/0xy9en/
/vi/firefox/user/0xy9en/
/ar/firefox/user/0xy9en/
/es/firefox/user/0xy9en/
/ru/firefox/user/0xy9en/
/en-US/firefox/user/00crimson/
/pt-BR/firefox/user/000000000o/
/zh-CN/android/user/0l1v3r/
/ja/android/user/0langy/
/zh-TW/android/user/0xSobky/
/ru/firefox/user/0xy9en/?src=api
/zh-CN/android/user/0bruno/?src=addon-detail
/uk/firefox/user/0l1v3r/?src=addon-detail
```

### What did you expect to happen?

User's profile pages do NOT return 5xx for certain users.

 
### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

**Comment by diox** [comment]
@AlexandraMoga added in the original issue:

> As far as I can tell, this is caused by the fact that the username starts with a `0` (see list above). I've experimented with a few more users and each time the username begins with a `0` the profile page will return a 504.
>
> Also, this seems to be happening only on the frontend, not in the API:
>
> https://addons-dev.allizom.org/api/v3/accounts/account/0basket/ - returns 200
> https://addons-dev.allizom.org/en-US/firefox/user/0basket/ - returns 504

**Comment by willdurand** [comment]
```js
> parseInt('0basket', 10)
0

> Number.isNaN(parseInt('0basket', 10))
false
```

**Comment by willdurand** [comment]
Because of the JS wtf above, parsing a string that is obviously not a number returns a number that is `0`. We do not accept `0` as a valid `userId` so we throw an error _via_ `invariant`, which is the cause of the 500.

Here is a fix:

```diff
diff --git a/src/amo/components/UserProfile/index.js b/src/amo/components/UserProfile/index.js
index 10c6f54..706467c 100644
--- a/src/amo/components/UserProfile/index.js
+++ b/src/amo/components/UserProfile/index.js
@@ -448,7 +448,7 @@ export function mapStateToProps(state: AppState, ownProps: Props) {
   // `getUserByUsername()` requires a string as second argument.
   let user = getUserByUsername(state.users, `${username}`);
 
-  if (!user) {
+  if (!user && /^[0-9]+$/.test(username)) {
     const userId = parseInt(username, 10);
     user = !Number.isNaN(userId) ? getUserById(state.users, userId) : undefined;
   }
```

**Comment by diox** [comment]
In case anyone was wondering, it appears to be working fine for add-ons. See for instance https://addons-dev.allizom.org/en-US/firefox/addon/0180-telefonbuch/

**Comment by kumar303** [comment]
@willdurand thanks, I think your strategy is the best one. Changing `getByUserId()` in the reducer to allow `0` as input only fixes the problem by way of a side effect.

**Comment by kumar303** [comment]
<img width="219" alt="screenshot 2018-08-22 15 59 01" src="https://user-images.githubusercontent.com/55398/44490622-5df26400-a624-11e8-84d1-0db79e9a9949.png">

ZOMG this was the [intended behavior](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description).


> If parseInt encounters a character that is not a numeral in the specified radix, it ignores it and all succeeding characters and returns the integer value parsed up to that point. 

🕺 🌀

**Comment by AlexandraMoga** [comment]
I have tried various scenarios with usernames starting with a `0` and I couldn't trigger an error any longer.

https://addons-dev.allizom.org/en-US/firefox/user/0ba123-_/ should load without any errors

The only way I could trigger a 5xx (I've got 500 and 504) was with a username having only `0s` - i.e. https://addons-dev.allizom.org/en-US/firefox/user/00000000000/
 I used Admin Tools to create that username, so I don't think this is a case that should ever happen in a real scenario, mainly because the frontend and the API prohibit digit only usernames.

@kumar303 let me know if you have any concerns about this issue
